src/main.cpp

```diff
@@ -410,6 +419,32 @@ void DaemonCoreApplication::serverReceived(QTcpSocket *socket, const QJsonObject
 
 		server.send(socket, response);
 
+	} else if (command == "load_config") {
+		if (!this->hasWriteAccess(socket))
+			return sendAccessDenied(socket, request);
+
+		QString filename = this->configFileName;
+		if (request.contains("filename"))
+			filename = request["filename"].toString();
+		bool ok = true;
+		try {
+			this->loadConfig(filename);
```

Comment:
Tohle teda funguje tak, že se úplně zapomene stav modulu a vyrobí se nový modul, že? Co když má někdo modul zaregistrovaný k informování o změnách? Co když například na sběrnici je momentálně aktivní modul jiného typu, než je načten? Co když v modulu momentálně probíhá nějaká operace, třeba aktualizace FW? To může asi skončit docela nepěkným segfaultem v nějakém callbacku, protože "this" přestane být platný pointer, ne?

Nejsem přesvědčen, že tento způsob implementace opětovného načtení konfigurace je v pořádku.

Reply:
Když má modul nějaký klient zaregistrovaný, vše proběhne v pořádku – registrace jsou mimo třídu desky a jsou vázané pouze na adresu (to by totiž nefungovalo nahrazení modulu při načtení jiného typu, než je v konfiguraci). Ostatní otázky ale trvají.

Reply:
Ale to je tvoje divná implementace funkce loadConfig co se ti nelíbí. Za mě to mám udělané stejně jak to děláš ty.

Reply:
To je možné, ale je fakt, že reload konfigurace za provozu prostě nejde jen takhle jednoduše udělat. To je nejspíš také důvod, proč jsem ho již neimplementoval. Máš to udělané stejně, ale moje funkce `loadConfig` počítá s tím, že se volá jen na začátku. Nepočítá s tím, že ji někdo bude volat, když už je aplikace v provozu.

Nevím, jak rozumně tenhle problém vyřešit. Asi by bylo třeba zajistit, aby při rušení instance modulu nebyl nikde uložený `this` ukazatel - zejména v žádném capture "aktivní" lambda funkce.